chromium_src/chrome/browser/ui/webui/settings/settings_secure_dns_handler.cc

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/ui/webui/settings/settings_secure_dns_handler.h"
+#include "base/feature_list.h"
+#include "base/values.h"
+#include "brave/components/brave_vpn/buildflags/buildflags.h"
+#include "build/buildflag.h"
+#include "chrome/common/pref_names.h"
+#include "components/prefs/pref_service.h"
+
+#if BUILDFLAG(IS_WIN) && BUILDFLAG(ENABLE_BRAVE_VPN)
+#include "brave/components/brave_vpn/features.h"
+
+#define kDnsOverHttpsTemplates kDnsOverHttpsTemplates, \
```

Comment:
what is this needed for?

Reply:
this is for the case when we have the settings page opened and enabling vpn, subscription for the pref to refresh state on the page to lock it if required

Reply:
like other overrides we've discussed, this is fragile and hard to read and understand and could be accomplished by subclassing `SecureDnsPolicyHandler` and overriding settings_ui.cc. We don't care about privacy_section.cc because that is chromeos.

Reply:
done here https://github.com/brave/brave-core/pull/15989/commits/c38356a03c3740efeed32d75b23cd7eeca2588ec